src/main/java/com/mathworks/ci/SourceFolder.java

```diff
@@ -0,0 +1,54 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2019-2020 The MathWorks, Inc.
+ *
+ * Describable class for Source Folder Option in RunMATLABTest Build step.
+ *
+ */
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SourceFolder extends AbstractDescribableImpl<SourceFolder> {
+
+    private List<SourceFolderPaths> sourceFolderPaths = new ArrayList<>();
+    private static final String SOURCE_FOLDER = "SourceFolder";
+
+    @DataBoundConstructor
+    public SourceFolder() {
+
+    }
+
+    @DataBoundSetter
+    public void setSourceFolderPaths(List<SourceFolderPaths> sourceFolderPaths) {
+        this.sourceFolderPaths = Util.fixNull(sourceFolderPaths);
+    }
+
+    public List<SourceFolderPaths> getSourceFolderPaths() {
+        return this.sourceFolderPaths;
+    }
+
+    public  void addFilePathArgTo(Map<String, String> inputArgs) {
+        // Concatenate all source folders to a single ";" separated string
+        inputArgs.put(SOURCE_FOLDER, this.sourceFolderPaths.stream()
+                .map(SourceFolderPaths::getSrcFolderPath)
+                .collect(Collectors.joining(";")));
```

Comment:
Just note that using this approach if someone provides a path including a colon, like with an absolute Windows path including a drive letter (e.g. C:\my\folder), the folder will be split into two (e.g. "C" and "\my\folder"). I know the help text says you must use a relative path but I think some users are bound to try an absolute path anyway.

We accepted this limitation previously because the only way a user could provide multiple paths in other systems is with a single string and we wanted to support both ":" and ";" as a path separator. With Jenkins, we don't have that limitation, so perhaps we should construct a cell array instead?

Reply:
Thank you for pointing this out. Refactored the code to generate a string in cell array format.